ipv6/prefix_test.go

```diff
@@ -0,0 +1,380 @@
+package ipv6
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func _p(cidr string) Prefix {
+	prefix, err := ParsePrefix(cidr)
+	if err != nil {
+		panic("only use this is happy cases")
+	}
+	return prefix
+}
+
+func unsafePrefixFromUint64(high, low uint64, length int) Prefix {
+	mask, err := MaskFromLength(length)
+	if err != nil {
+		panic("only use this is happy cases")
+	}
+	return PrefixFromAddressMask(Address{uint128{high, low}}, mask)
+}
+
+func unsafeParseNet(prefix string) *net.IPNet {
+	ipNet, err := parseNet(prefix)
+	if err != nil {
+		panic("only use this is happy cases")
+	}
+	return ipNet
+}
+
+func TestPrefixComparable(t *testing.T) {
+	tests := []struct {
+		description string
+		a, b        Prefix
+		equal       bool
+	}{
+		{
+			description: "equal",
+			a:           _p("2001::1/64"),
+			b:           _p("2001::1/64"),
+			equal:       true,
+		}, {
+			description: "lengths not equal",
+			a:           _p("2001::1/64"),
+			b:           _p("2001::1/65"),
+			equal:       false,
+		}, {
+			description: "host bits not equal",
+			a:           _p("2001::1/64"),
+			b:           _p("2001::2/64"),
+			equal:       false,
+		}, {
+			description: "prefix bits not equal",
+			a:           _p("2001::1/64"),
+			b:           _p("2002::1/64"),
+			equal:       false,
+		}, {
+			description: "extremes",
+			a:           _p("::/0"),
+			b:           _p("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128"),
+			equal:       false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			assert.Equal(t, tt.equal, tt.a == tt.b)
+			assert.Equal(t, !tt.equal, tt.a != tt.b)
+		})
+	}
+}
+
+func TestParsePrefix(t *testing.T) {
+	tests := []struct {
+		description string
+		cidr        string
+		expected    Prefix
+		isErr       bool
+	}{
+		{
+			description: "success",
+			cidr:        "10.224.24.1/27",
```

Comment:
Both this and the address unit tests should have a case that tests parsing an IPv6 address in IPv4-mapped notation. For example, the following is a valid IPv6 address, [believe it or not](https://stackoverflow.com/questions/166132/maximum-length-of-the-textual-representation-of-an-ipv6-address#166157):

```
::ffff:203.0.113.17
```

If I remember correctly, the underlying parser can handle it but I'm not 100% on that.